enhancements/api-review/add-ClusterImagePolicy-and-ImagePolicy-for-Signature-Verification.md

```diff
@@ -192,6 +197,38 @@ type FulcioCAWithRekor struct {
 	FulcioSubject PolicyFulcioSubject `json:"fulcioSubject,omitempty"`
 }
 
+// PKI defines the root of trust based on Root CA(s) and corresponding intermediate certificates.
+type PKI struct {
+	// CertificateAuthorityRootsData contains base64-encoded data of a certificate bundle PEM file, which contains one or more CA roots
+	// in the PEM format.
+	// +required
+	CertificateAuthorityRootsData []byte `json:"caRootsData,omitempty"`
+
+	// CertificateAuthorityIntermediatesData base64-encoded data of a certificate bundle PEM file, which contains one or more intermediate certificates
+	// in the PEM format.
+	// caIntermediatesData requires CertificateAuthorityRoots to be set.
+	// +optional
+	// +kubebuilder:validation:XValidation:rule="self == '' || self.CertificateAuthorityRootsData != ''", message="caIntermediatesData requires caRootsData to be set"
+	CertificateAuthorityIntermediatesData []byte `json:"caIntermediatesData,omitempty"`
+
+	// PKICertificateSubject must be set if any of the other fields are set.
+	// +optional
+  // +kubebuilder:validation:XValidation:rule="self.CertificateAuthorityRootsData == '' || self.PKICertificateSubject != nil", message="pkiCertificateSubject must be set if caRootsData is specified"
+	// +kubebuilder:validation:XValidation:rule="self.Email != '' || self.Hostname != ''", message="At least one of Email or Hostname must be set in pkiCertificateSubject"
+	PKICertificateSubject *PKICertificateSubject `json:"pkiCertificateSubject,omitempty"`
+}
+
+// PKICertificateSubject defines the identity type associated with the certificate identity
```

Comment:
```suggestion
// PKICertificateSubject defines the requirements imposed on the subject to which the certificate was issued.
```

---

The fields below should also talk about subject, not identity — and perhaps explicitly point say that this  matches against the corresponding “subject alternative name” fields.

Reply:
Updated, thanks.

Reply:
Still s/certificate identity/certificate subject/ in the fields of `…CertficateSubject`, please.

Reply:
I just saw this review, I can address and let people review the API comments/validation/naming details in the openshift/api PR later. cc @harche